tests/test_utils/test_schemaview.py

```diff
@@ -1036,40 +1036,67 @@ def test_enums_and_enum_relationships(schema_view_no_imports: SchemaView) -> Non
-                    assert "EAGLE" not in view.permissible_value_descendants(pv, e.name)
-
-                if pv == "LION":
-                    assert "ANGRY_LION" in view.permissible_value_children(pv, e.name)
-
-                if pv == "ANGRY_LION":
-                    assert view.permissible_value_parent(pv, e.name) == ["LION"]
-                    assert view.permissible_value_ancestors(pv, e.name) == ["ANGRY_LION", "LION", "CAT"]
-                    assert view.permissible_value_descendants(pv, e.name) == ["ANGRY_LION"]
+    animals = "Animals"
+    animal_enum = view.get_enum(animals)
+    assert animal_enum.name == animals
+
+    pv_cat = animal_enum.permissible_values["CAT"]
+    assert pv_cat.text == "CAT"
+    assert pv_cat.is_a is None
+    assert view.permissible_value_parent("CAT", animals) is None
+    assert view.permissible_value_ancestors("CAT", animals) == ["CAT"]
+    assert set(view.permissible_value_children("CAT", animals)) == {"LION", "TABBY"}
+    assert set(view.permissible_value_descendants("CAT", animals)) == {"CAT", "LION", "ANGRY_LION", "TABBY"}
+
+    pv_tabby = animal_enum.permissible_values["TABBY"]
+    assert pv_tabby.is_a == "CAT"
+    assert view.permissible_value_parent("TABBY", animals) == ["CAT"]
+    assert view.permissible_value_ancestors("TABBY", animals) == ["TABBY", "CAT"]
+    assert view.permissible_value_children("TABBY", animals) == []
+    assert view.permissible_value_descendants("TABBY", animals) == ["TABBY"]
+
+    pv_lion = animal_enum.permissible_values["LION"]
+    assert pv_lion.is_a == "CAT"
+    assert view.permissible_value_parent("LION", animals) == ["CAT"]
+    assert view.permissible_value_ancestors("LION", animals) == ["LION", "CAT"]
+    assert view.permissible_value_children("LION", animals) == ["ANGRY_LION"]
+    assert view.permissible_value_descendants("LION", animals) == ["LION", "ANGRY_LION"]
+
+    pv_angry_lion = animal_enum.permissible_values["ANGRY_LION"]
+    assert pv_angry_lion.is_a == "LION"
+    assert view.permissible_value_parent("ANGRY_LION", animals) == ["LION"]
+    assert view.permissible_value_ancestors("ANGRY_LION", animals) == ["ANGRY_LION", "LION", "CAT"]
+    assert view.permissible_value_children("ANGRY_LION", animals) == []
+    assert view.permissible_value_descendants("ANGRY_LION", animals) == ["ANGRY_LION"]
+
+
+# FIXME: improve testing of dynamic enums
+def test_dynamic_enum(schema_view_with_imports: SchemaView) -> None:
+    """Rudimentary test of dynamic enum."""
+    view = schema_view_with_imports
 
-    for cn, c in view.all_classes().items():
-        if c.name == "Adult":
-            assert view.class_ancestors(c.name) == ["Adult", "Person", "HasAliases", "Thing"]
+    # dynamic enums
+    e = view.get_enum("HCAExample")
+    assert set(e.include[0].reachable_from.source_nodes) == {"GO:0007049", "GO:0022403"}
 
 
-def test_schemaview(schema_view_no_imports: SchemaView) -> None:
```

Comment:
This test is far too long and tests too many different things. It is better to keep tests short and focused on specific, related methods. I went through the method and split it up as logically as I could.